test/hotspot/jtreg/compiler/vectorapi/VectorMultiplyOpt.java

```diff
@@ -0,0 +1,230 @@
+        IntStream.range(0, SIZE).forEach(i -> { lsrc2[i] = Long.MAX_VALUE * r.nextLong(); });
+        IntStream.range(0, SIZE).forEach(i -> { isrc1[i] = Integer.MAX_VALUE * r.nextInt(); });
+        IntStream.range(0, SIZE).forEach(i -> { isrc2[i] = Integer.MAX_VALUE * r.nextInt(); });
+    }
+
+
+    public static void main(String[] args) {
+        TestFramework testFramework = new TestFramework();
+        testFramework.setDefaultWarmup(5000)
+                     .addFlags("--add-modules=jdk.incubator.vector")
+                     .start();
+        System.out.println("PASSED");
+    }
+
+    interface Validator {
+        public long apply(long src1, long src2);
+    }
+
+    public static void validate(String msg, long[] actual, Object src1, Object src2, Validator func) {
+        for (int i = 0; i < actual.length; i++) {
+            long expected;
+            if (long[].class == src1.getClass()) {
+                expected = func.apply(Array.getLong(src1, i), Array.getLong(src2, i));
+            } else {
+                assert int[].class == src1.getClass();
+                expected = func.apply(Array.getInt(src1, i), Array.getInt(src2, i));
+            }
+            if (actual[i] != expected) {
+                throw new AssertionError(msg + "index " + i + ": src1 = " + Array.get(src1, i) + " src2 = " +
+                                         Array.get(src2, i) + " actual = " + actual[i] + " expected = " + expected);
+            }
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.MUL_VL, " >0 ", IRNode.AND_VL, " >0 "}, applyIfCPUFeature = {"avx", "true"})
+    @Warmup(value = 10000)
+    public static void test_pattern1() {
+        int i = 0;
+        for (; i < LSP.loopBound(res.length); i += LSP.length()) {
+            LongVector vsrc1 = LongVector.fromArray(LSP, lsrc1, i);
+            LongVector vsrc2 = LongVector.fromArray(LSP, lsrc2, i);
+            vsrc1.lanewise(VectorOperators.AND, 0xFFFFFL)
+                 .lanewise(VectorOperators.MUL, vsrc2.lanewise(VectorOperators.AND, 0xFFFFFL))
+                 .intoArray(res, i);
+        }
+        for (; i < res.length; i++) {
+            res[i] = (lsrc1[i] & 0xFFFFFL) * (lsrc2[i] & 0xFFFFFL);
+        }
+    }
+
+    @Check(test = "test_pattern1")
+    public void test_pattern1_validate() {
+        validate("pattern1 ", res, lsrc1, lsrc2, (l1, l2) -> (l1 & 0xFFFFFL) * (l2 & 0xFFFFFL));
+    }
+
+    @Test
+    @IR(counts = {IRNode.MUL_VL, " >0 ", IRNode.AND_VL, " >0 ", IRNode.URSHIFT_VL, " >0 "}, applyIfCPUFeature = {"avx", "true"})
+    @Warmup(value = 10000)
+    public static void test_pattern2() {
+        int i = 0;
+        for (; i < LSP.loopBound(res.length); i += LSP.length()) {
+            LongVector vsrc1 = LongVector.fromArray(LSP, lsrc1, i);
+            LongVector vsrc2 = LongVector.fromArray(LSP, lsrc2, i);
+            vsrc1.lanewise(VectorOperators.AND, 0xFFFFFFL)
```

Comment:
Alternatively, you could populate the constants in randomized manner and put them into static final fields during class initialization. Then just load them from there in test code.